File: javascript/hamming/hamming.js
module.exports = function Hamming() {
    return {
        compute(a, b) {
            if (a.length !== b.length) { throw new Error('DNA strands must be of equal length.'); }
            b = b.split('');
            return a.split('')
                .map((char, index) => char !== b[index] ? 1 : 0)
                .reduce((a, b) => a + b);
        }
    }
};

File: java/hamming/src/main/java/Hamming.java
import java.util.stream.IntStream;
import java.util.Arrays;


public class Hamming {
    String left;
    String right;

    Hamming(String leftStrand, String rightStrand) {
        if (leftStrand.length() != rightStrand.length()) {
            throw new IllegalArgumentException("leftStrand and rightStrand must be of equal length.");
        }
        left = leftStrand;
        right = rightStrand;
    }

    int getHammingDistance() {
        return IntStream.range(0, left.length())
            .reduce(0, (acc, i) ->
                left.charAt(i) == right.charAt(i) ? acc : acc + 1
            );
    }

}
